src/modules/txbuilder/RadixTransactionBuilder.ts

```diff
@@ -383,41 +383,115 @@ export default class RadixTransactionBuilder {
         return this
     }
 
-    public createToken(
+    public createTokenSingleIssuance(
         owner: RadixAccount,
         name: string,
         symbol: string,
         description: string,
-        granularity: number | string | Decimal,
-        decimalQuantity: number | string | Decimal,
+        granularity: string | number | Decimal = new Decimal('1e-18'),
+        decimalQuantity: string | number | Decimal,
         iconUrl: string,
-        permissions: RadixTokenPermissions,
     ) {
-        const tokenAmount = this.getSubUnitsQuantity(decimalQuantity)
-        const tokenGranularity = this.getSubUnitsQuantity(granularity)
+        const subunitsQuantity = this.getSubUnitsQuantity(decimalQuantity)
+        const subunitsGranularity = this.getSubUnitsQuantity(granularity)
+
+        if (subunitsQuantity.lte(this.BNZERO)) {
+            throw new Error('Single-issuance tokens must be created with some supply')
```

Comment:
Why not put this in the constructor of `RadixFixedSupplyTokenDefinitionParticle` instead, as per my previous comment :)